ts/Core/Axis/Axis.ts

```diff
@@ -407,11 +409,14 @@ class Axis {
          * @type {Highcharts.AxisOptions}
          */
         axis.setOptions(userOptions);
-
+        axis.uniqueNames ??= axis.options.uniqueNames ?? true;
```

Comment:
Maybe move this down to line 418 together with the equivalent `axis.type` assignment?